data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/ProcessWorker.java

```diff
@@ -126,13 +125,17 @@ private void doRun() {
         }
         //Should Empty list from buffer should be sent to the processors? For now sending as the Stateful processors expects it.
         for (final Processor processor : processors) {
-            List<Event> inputEvents = null;
-            if (acknowledgementsEnabled) {
-                inputEvents = ((ArrayList<Record<Event>>)records).stream().map(Record::getData).collect(Collectors.toList());
-            }
-            records = processor.execute(records);
-            if (inputEvents != null) {
-                processAcknowledgements(inputEvents, records);
+            try {
+                List<Event> inputEvents = null;
+                if (acknowledgementsEnabled) {
+                    inputEvents = ((ArrayList<Record<Event>>) records).stream().map(Record::getData).collect(Collectors.toList());
+                }
+                records = processor.execute(records);
+                if (inputEvents != null) {
+                    processAcknowledgements(inputEvents, records);
+                }
+            } catch (final Exception e) {
```

Comment:
Yes we could make it configurable to send these records to a pipeline DLQ. However, I am not sure if dropping the data is the best behavior here. As a user, I would rather get my data and not have it be as I expected rather than silently dropping data. Then, if it does not appear correctly in the sink, these error logs will help guide what the issue is. I tend to think that downstream processors should be robust enough to at least not crash when they do not get what they expect, even if it means they are tagged with certain values indicating failure to perform an action on the Event in the downstream processor

Reply:
I think for most processors, this is probably the case. Which is why I support adding the try-catch blocks in those processors.

However, some processors will not want this behavior. One example is the service map. If an exception leaks from there we'd be sending trace events to the service map index. Other examples are pipelines that generate metrics from logs and alerting on anomalies.

But, if we make this change in core, we force all events to flow through all processors this way.

I think the ideal solution has:
1. Appropriate try-catch in the processors where it is ok to do this.
2. Dropping events from exceptions in core

Reply:
That is a fair point I did not consider stateful processors. I am ok with changing to drop the batch instead of passing them downstream